pkg/cmd/server/etcd/etcd.go

```diff
@@ -133,3 +135,19 @@ func TestEtcdClient(etcdClient *etcdclient.Client) error {
 	}
 	return nil
 }
+
+// NeverRetryOnFailure is a retry function for the etcdClient.  If there's only one machine, master election doesn't make much sense,
+// so we don't bother to retry, we simply dump the failure and return the error directly.
+func NeverRetryOnFailure(cluster *etcdclient.Cluster, numReqs int, lastResp http.Response, err error) error {
+	if len(cluster.Machines) > 1 {
```

Comment:
> I don't disagree, but this is going to make things worse before they get better... it means that GET retries that are currently succeeding will now fail

You'd think, but see the chain from https://github.com/openshift/origin/pull/6529#issuecomment-169093790.  The errors that we're currently getting are only possible on mutation requests.  You' saying that you suspect we've gotten read errors before, but they've been different read errors that have always been recoverable?  I'd claim that's thin, but I will admit its possible.